src/pg_to_evalscript/evalscript.py

```diff
@@ -120,8 +134,14 @@ def write_update_output(self):
         size_without_original_temporal_dimensions = reduce(
             lambda x, y: x * y, sizes_without_original_temporal_dimensions, 1
         )
-        collection_scenes_length = "* collection.scenes.length" * number_of_original_temporal_dimensions
-        number_of_final_dimensions = len(self._output_dimensions) + 1 if self.encode_result else 0
+        if len(self.input_bands) > 1:
+            collection_scenes_length = (
+                "* Object.values(collection).reduce((acc, val) => acc + val.scenes.length, 0)"
+                * number_of_original_temporal_dimensions
+            )
+        else:
+            collection_scenes_length = "* collection.scenes.length" * number_of_original_temporal_dimensions
+        number_of_final_dimensions = len(self._output_dimensions) + 1 if self.encode_result else 0  # 3
```

Comment:
ah, missed this comment - it can probably be deleted

Reply:
removed